controllers/teleport_controller/teleport_controller.cpp

```diff
@@ -0,0 +1,138 @@
+static const int rotationsSize = 8;                                     
+static const double rotations[rotationsSize][4] = {{1, 0, 0, 1.57081},
+                                                  {0.867737, 0.343283, 0.359429, 1.75753},
+                                                  {0.577353, 0.577345, 0.577353, 2.0944},
+                                                  {0.286947, 0.661632, 0.692752, 2.60662},
+                                                  {1.69526e-09, 0.707103, 0.707111, 3.14159},
+                                                  {-0.286948, 0.661631, 0.692752, -2.60661},
+                                                  {-0.577352, 0.577347, 0.577352, -2.09441},
+                                                  {0.867739, -0.343277, -0.359429, 1.75753}};                               
+
+
+/* This is the main program of the supervisor. It creates an instance of the Supervisor
+ *     instance, launches its function(s) and destroys it at the end of the execution.
+ */
+int main() {
+    std::cout << "a" << std::endl;
+    // create the Supervisor instance and assign it to a robot
+    Supervisor* supervisor = new Supervisor();
+    Node* target           = supervisor->getFromDef("RED_1");
+    std::cout << "b" << std::endl;
+    // Get the time step of the current world.
+    int timeStep = (int) supervisor->getBasicTimeStep();
+
+    std::cout << "c" << std::endl;
+
+    // Add the motion manager to make the robot stand up and look around
+    managers::RobotisOp2MotionManager* mMotionManager = new managers::RobotisOp2MotionManager(supervisor);
+    mMotionManager->playPage(54);
+
+    // Generate random seed
+    srand(time(NULL));
+
+    // Main loop:
+    while (supervisor->step(timeStep) != -1) {
+
+        // Grab the current translation field of the robot to modify
+        Field* target_translation_field = target->getField("translation");
+        // Convert the field to a vector to output to console
+        const double* target_translation_vec = target_translation_field->getSFVec3f();
+
+        // Output current location
+        std::cout << "Location: " << std::endl;
+        std::cout << "X: " << target_translation_vec[0];
+        std::cout << " Y: " << target_translation_vec[1];
+        std::cout << " Z: " << target_translation_vec[2] << std::endl;
+
+        // Prepare new location
+
+        /* 0, 0, 0 is centre of the playing field.
+         *     X ranges from -5.4 - 5.4, the positive value on the left hand side when looking
+         *     from red
+         *     Y ranges from -8 - 8, the positive value on the red goal side
+         *     Z should be set to 3.2 to be on the ground
+         *     If the robot teleports into an existing object it may run into issues for that
+         *     image only, after resetPhysics is should return to a regular state
+         */
+
+        double newPos[3];
+        newPos[0] = 5.4 - (rand() % 1080) / 100;  // target_translation_vec[0] + 1;
```

Comment:
I don't understand what these end of line comments are saying

Reply:
Leftover comment from testing, will remove